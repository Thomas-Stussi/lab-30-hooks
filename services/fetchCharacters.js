/* eslint-disable max-len */
export const fetchCharacters = () => {
  return fetch('https://xfiles-api.herokuapp.com/api/v1/characters')
    .then(res => res.json())
    .then(json => json.results);
};

export const fetchCharacter = (name) => {
  return fetch(`https://xfiles-api.herokuapp.com/api/v1/characters${name}`)
    .then(res => res.json())
    .then(json => json[0]);
};
