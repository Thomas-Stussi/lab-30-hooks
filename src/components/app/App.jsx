/* eslint-disable max-len */
import React from 'react';
import Xfiles from '../../containers/Xfiles';
import XfilesDetails from '../../containers/XfilesDetails';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';

export default function App() {
  return <div>
    <h1>The Truth is Here</h1>
    <Router>
      <Switch>
        <Route
          path="/"
          exact
          render={() => <Xfiles />}
        />
        <Route
          path="/:name"
          exact
          render={() => <XfilesDetails />}
        />
      </Switch>
    </Router>
  </div>;
}
