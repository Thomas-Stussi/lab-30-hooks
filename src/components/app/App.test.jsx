/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { fetchCharacters } from '../../../services/fetchCharacters';

jest.mock('../../../services/fetchCharacters.js');

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    fetchCharacters.mockResolvedValue([
      { name: 'test1', image: 'https://www.placecage.com/c/140/200' }, { name: 'test2', image: 'https://www.placecage.com/c/140/200' }]);
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
