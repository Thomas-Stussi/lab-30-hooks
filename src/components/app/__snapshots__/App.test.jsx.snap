// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <h1>
      The Truth is Here
    </h1>
    <div>
      <p>
        Loading
      </p>
      <img
        data-testid="loading"
        src="https://i.giphy.com/media/uvDn0BLHElgRO/giphy.webp"
      />
    </div>
  </div>
</DocumentFragment>
`;
